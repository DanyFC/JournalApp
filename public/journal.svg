<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" width="100" height="100" viewBox="0 0 48 48">
<path fill="#3dd9eb" d="M40,39H8V9c0-2.2,1.8-4,4-4h28V39z"></path><rect width="18" height="9" x="15" y="15" fill="#7debf5"></rect><path fill="#3dd9eb" d="M8,9v30h4V5C9.8,5,8,6.8,8,9z"></path><path fill="#7debf5" d="M40,43H11.5C9.6,43,8,41.4,8,39.5l0,0c0-1.9,1.6-3.5,3.5-3.5H40V43z"></path><path fill="#00b3d7" d="M8.1,39h32v-3H11.5C9.7,36,8.3,37.3,8.1,39z"></path><polygon fill="#eb0000" points="23,2 23,28 29.5,25 36,28 36,2"></polygon><rect width="13" height="3" x="23" y="2" fill="#f55376"></rect>
</svg>