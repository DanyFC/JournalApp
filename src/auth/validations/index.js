export { default as loginValidation } from './loginValidations'
export { default as registerValidation } from './registerValidation'