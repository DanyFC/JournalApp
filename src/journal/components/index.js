export { default as Gallery } from './Gallery'
export { default as Navbar } from './Navbar'
export { default as Sidebar } from './Sidebar'