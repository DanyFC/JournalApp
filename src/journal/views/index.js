export { default as Note } from './Note'
export { default as NothingSelected } from './NothingSelected'

