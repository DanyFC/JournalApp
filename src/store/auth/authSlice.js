import { createSlice } from '@reduxjs/toolkit'

export const authSlice = createSlice({
  name: 'auth',
  initialState: {
    displayName: null,
    email: null,
    errorMessage: null,
    photoURL: null,
    status: 'checking', // INFO: 'checking', 'not-authenticated', 'authenticated'
    uid: null
  },
  reducers: {
    checkCredential: (state) => {
      state.status = 'checking'
    },
    login: (state, { payload }) => {
      state.displayName = payload.displayName
      state.email = payload.email
      state.errorMessage = null
      state.photoURL = payload.photoURL
      state.status = 'authenticated'
      state.uid = payload.uid
    },
    logout: (state, { payload }) => {
      state.displayName = null
      state.email = null
      state.errorMessage = payload ? payload : null
      state.photoURL = null
      state.status = 'not-authenticated'
      state.uid = null
    }
  },
})

export const { checkCredential, login, logout } = authSlice.actions