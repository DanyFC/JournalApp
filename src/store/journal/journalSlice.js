import { createSlice } from '@reduxjs/toolkit'

export const journalSlice = createSlice({
  name: 'journal',
  initialState: {
    isSaving: false,
    savedMessage: null,
    notes: [],
    activeNote: null // INFO: {id: string, title: string, body: string, date: Date, imageUrls: array}
  },
  reducers: {
    addNewEmptyNote: (state, action) => {
      state.notes.push(action.payload)
      state.isSaving = false
    },
    clearState: (state) => {
      state.isSaving = false
      state.savedMessage = null
      state.notes = []
      state.activeNote = null
    },
    deleteNoteById: (state, action) => {
      state.notes = state.notes.filter(note => note.id !== action.payload)
      state.savedMessage = `The note was successfully deleted.`
      state.isSaving = false
    },
    setActiveNote: (state, action) => {
      state.activeNote = action.payload
    },
    setPhotosToActiveNote: (state, action) => {
      state.activeNote.imageUrls = [...state.activeNote.imageUrls, ...action.payload]
      state.isSaving = false
    },
    setNotes: (state, action) => {
      state.notes = action.payload
      state.isSaving = false
    },
    setSaving: (state) => {
      state.isSaving = true
      state.savedMessage = null
    },
    updateNote: (state, action) => {
      const { id, title, body, imageUrls } = action.payload
      state.notes = state.notes.map(note => {
        if (note.id === id) return { ...note, title, body, imageUrls }
        return note
      })
      state.savedMessage = `The note "${title}" was successfully updated.`
      state.isSaving = false
    }
  },
})

export const {
  addNewEmptyNote,
  clearState,
  deleteNoteById,
  setActiveNote,
  setPhotosToActiveNote,
  setNotes,
  setSaving,
  updateNote
} = journalSlice.actions