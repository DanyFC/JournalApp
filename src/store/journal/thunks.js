import { addNewEmptyNote, deleteNoteById, setActiveNote, setNotes, setPhotosToActiveNote, setSaving, updateNote } from './'
import { collection, deleteDoc, doc, getDocs, setDoc } from 'firebase/firestore/lite'
import { fileUpload } from '../../helpers'
import { FirebaseDB } from '../../firebase/config'

export const startNewNote = () => {
  return async (dispatch, getState) => {
    dispatch(setSaving())

    const { uid } = getState().auth
    const newNote = {
      title: '',
      body: '',
      date: new Date().getTime(),
      imageUrls: []
    }

    const newDoc = doc(collection(FirebaseDB, `${uid}/journal/notes`))
    await setDoc(newDoc, newNote)
    newNote.id = newDoc.id

    dispatch(addNewEmptyNote(newNote))
    dispatch(setActiveNote(newNote))
  }
}

export const startLoadingNotes = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth
    const collectionRef = collection(FirebaseDB, `${uid}/journal/notes`)
    const querySnapshot = await getDocs(collectionRef)
    const notes = querySnapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }))

    dispatch(setNotes(notes))
  }
}

export const startUpdateNote = (note) => {
  return async (dispatch, getState) => {
    dispatch(setSaving())

    const { uid } = getState().auth
    const noteRef = doc(FirebaseDB, `${uid}/journal/notes/${note.id}`)
    const noteToFirestore = { ...note }
    delete noteToFirestore.id
    await setDoc(noteRef, noteToFirestore, { merge: true })

    dispatch(updateNote(note))
  }
}

export const startUploadingFiles = (note, files = []) => {
  return async (dispatch) => {
    dispatch(setSaving())

    const fileUploadPromises = []
    for (const file of files) fileUploadPromises.push(fileUpload(file))
    const photosUrls = await Promise.all(fileUploadPromises)

    dispatch(setActiveNote(note))
    dispatch(setPhotosToActiveNote(photosUrls))
  }
}

export const startDeleteNote = () => {
  return async (dispatch, getState) => {
    dispatch(setSaving())
    const { uid } = getState().auth
    const { id } = getState().journal.activeNote
    const noteRef = doc(FirebaseDB, `${uid}/journal/notes/${id}`)
    await deleteDoc(noteRef)
    dispatch(deleteNoteById(id))
    dispatch(setActiveNote(null))
  }
}