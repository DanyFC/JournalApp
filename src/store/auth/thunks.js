import { checkCredential, login, logout } from './'
import { loginWithEmailPassword, logoutFirebase, registerUserWithEmailPassword, signInWithGoogle } from '../../firebase/providers'
import { clearState } from '../journal'

export const startLoginWithEmailPassword = (user) => {
  return async (dispatch) => {
    dispatch(checkCredential())
    const result = await loginWithEmailPassword(user)
    if (!result.ok) {
      dispatch(logout(result.errorMessage))
      return
    }
    dispatch(login(result))
  }
}

export const startGoogleSignIn = () => {
  return async (dispatch) => {
    dispatch(checkCredential())
    const result = await signInWithGoogle()
    if (!result.ok) {
      dispatch(logout(result.errorMessage))
      return
    }
    dispatch(login(result))
  }
}

export const startCreatingUserWithEmailPassword = (newUser) => {
  return async (dispatch) => {
    dispatch(checkCredential())
    const result = await registerUserWithEmailPassword(newUser)
    if (!result.ok) {
      dispatch(logout(result.errorMessage))
      return
    }
    dispatch(login(result))
  }
}

export const startLogout = () => {
  return async (dispatch) => {
    await logoutFirebase()
    dispatch(clearState())
    dispatch(logout())
  }
}