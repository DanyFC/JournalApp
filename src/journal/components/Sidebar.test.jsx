/**
 * @jest-environment jsdom
*/
import { activeNoteState } from '../../fixtures/journal'
import { authenticatedState } from '../../fixtures/auth'
import { authSlice } from '../../store/auth'
import { configureStore } from '@reduxjs/toolkit'
import { journalSlice } from '../../store/journal'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import Sidebar from './Sidebar'

const store = configureStore({
  reducer: {
    auth: authSlice.reducer,
    journal: journalSlice.reducer
  },
  preloadedState: {
    auth: authenticatedState,
    journal: activeNoteState
  }
})

/* eslint-disable no-undef */
describe('Tests in <Sidebar />', () => {

  test('Can be match with snapshot.', () => {
    const { container } = render(
      <Provider store={store}>
        <Sidebar drawerWidth={200} />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
/* eslint-enable no-undef */
