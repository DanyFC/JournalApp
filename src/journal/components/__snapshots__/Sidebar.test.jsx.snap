// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Sidebar /> Can be match with snapshot. 1`] = `
<div>
  <nav
    class="MuiBox-root css-obqxoz"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1xkgjv5-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
          >
            userTest
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <ul
          class="MuiList-root MuiList-padding animate__animated animate__slideInLeft animate__delay-1s css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TurnedInNotIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2m0 15-5-2.18L7 18V5h10z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    First note
                  </span>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
                  >
                    This is a first note
                  </p>
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TurnedInNotIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2m0 15-5-2.18L7 18V5h10z"
                  />
                </svg>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Demo Note
                  </span>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
                  >
                    This is a demo note
                  </p>
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
